php/class-wp-cli.php

```diff
@@ -45,6 +45,145 @@ public function create_guest_authors( $args, $assoc_args ) {
+	 *
+	 * [--dry-run]
+	 * : By default, the command outputs the IDs of the guest authors it would have attempted to delete, had this not been a dry run. When you are ready to modify data, pass --no-dry-run.
+	 *
+	 * ## EXAMPLES
+	 *
+	 * # Dry run version of deleting guest authors.
+	 * $ wp co-authors-plus delete-guest-authors testFile.csv
+	 * DRY RUN
+	 * Attempting to delete guest author with ID 1, intended assignee smistephen
+	 *
+	 * # Actually deleting guest authors.
+	 * $ wp co-authors-plus delete-guest-authors testFile.csv --no-dry-run
+	 * You have provided the --no-dry-run flag. Please confirm you wish to delete guest authors. [y/n]
+	 * Successfully deleted guest author 1
+	 * OR
+	 * Error while attempting to delete guest author 1: Guest author does not exist.
+	 *
+	 * @subcommand delete-guest-authors
+	 * @synopsis <guest-author-ids-file> [--dry-run]
+	 */
+	public function delete_guest_authors( $args, $assoc_args ) {
+		global $coauthors_plus;
+
+		$defaults = array(
+			'dry-run' => true,
+		);
+
+		$assoc_args = wp_parse_args( $assoc_args, $defaults );
+
+		$dry_run = $assoc_args['dry-run'];
+
+		// Safety first.
+		$filename_status = validate_file( $args[0] );
+
+		if ( 0 !== $filename_status ) {
+			WP_CLI::error( __( 'The provided file cannot be accessed. Please use either relative or absolute pathnames, and avoid using directory traversal such as ../ and other methods.', 'co-authors-plus' ) );
+		}
+
+		$filename = $args[0];
+
+		// Add custom error handling on fopen error.
+		// Traditionally, fopen failures result in E_WARNING messages being logged to the
+		// console. But those are really ugly and disquieting for end users to see, and usually
+		// it makes them think something is wrong with the code, rather than the simple error
+		// of "this file isn't available / isn't readable by this user / etc".
+		// So I decided to pass the message through wp-cli's display API to make it look nicer.
+		set_error_handler(
+			function( $errno, $errstr, $errfile, $errline, array $errcontext ) {
+				WP_CLI::error( sprintf(
+					/* translators: %s: Error message */
+					__( 'Error while attempting to open file: %s', 'co-authors-plus' ),
+					$errstr
+				) );
+			},
+			E_WARNING
+		);
+
+		$file = fopen( $filename, 'rb' );
+
+		// Reset normal behaviour.
+		restore_error_handler();
+
+		// Are you really, really sure.
+		if ( false === $dry_run ) {
+			WP_CLI::confirm( __( 'You have provided the --no-dry-run flag. Please confirm you wish to delete guest authors.', 'co-authors-plus' ) );
+		} else {
+			WP_CLI::log( __( 'DRY RUN', 'co-authors-plus' ) );
+		}
+
+		while ( ( $line = fgetcsv( $file, 100, ',' ) ) !== false ) {
+			$id       = filter_var( $line[0], FILTER_SANITIZE_NUMBER_INT );
+			$assignee = filter_var( $line[1], FILTER_SANITIZE_STRING );
```

Comment:
What happens if this assignee doesn't exist?

Reply:
The function returns a WP_Error :+1: 
https://github.com/Automattic/Co-Authors-Plus/pull/588#discussion_r210121608

Reply:
I was referring to the value of `$assignee` - what happens if the assignee (i.e. the username) given in the CSV doesn't exist? What will this code do to that value?

Reply:
Ah yes. That is an error, and it has now been fixed.